src/sql/src/plan/statement/ddl.rs

```diff
@@ -546,19 +551,37 @@ pub fn plan_create_source(
                 .region
                 .ok_or_else(|| sql_err!("Provided ARN does not include an AWS region"))?;
 
-            let aws = normalize::aws_config(&mut legacy_with_options, Some(region.into()))?;
+            let item = scx.get_item_by_resolved_name(&aws_connection)?;
+            let aws_connection = match item.connection()? {
+                Connection::Aws(connection) => connection.clone(),
+                _ => sql_bail!("{} is not an AWS connection", item.name()),
+            };
```

Comment:
is this not guaranteed by parsing? can we just panic?

Reply:
Not guaranteed by parsing; it just gets the name of the object without checking its type.